src/name_service/core/use_case.clj

```diff
@@ -10,17 +10,61 @@
       (action/make-action type payload))))
 
 (defn add-key-binding [{:keys [storage]}]
-  "Add value 'val' for key 'key'."
-  (fn [key val]))
+  "Add value A for key X."
+  (fn [key val]
+    (let [keymap {key val}]
+      (result->action
+        :entity/add
+        (entity/save storage keymap nil nil)))))
+
+(defn append-key-binding [{:keys [storage]}]
+  "Add a value A for key X, where key Y has value B."
+  (fn [key-a val-a key-b val-b]
+    (let [keymap (entity/fetch storage key-b val-b)]
```

Comment:
To be perfectly clear, `keymap` should be label `keymap-or-error` but this is fine I think.